web/third/marked/lib/marked.js

```diff
@@ -1400,79 +1400,7 @@ function merge(obj) {
  * Marked
  */
 
-function marked(src, opt, callback) {
-  if (callback || typeof opt === 'function') {
-    if (!callback) {
-      callback = opt;
-      opt = null;
-    }
-
-    opt = merge({}, marked.defaults, opt || {});
-
-    var highlight = opt.highlight
-      , tokens
-      , pending
-      , i = 0;
-
-    htmlStashCounter = 0;
-    htmlStash = [];
-
-    try {
-      tokens = Lexer.lex(src, opt)
-    } catch (e) {
-      return callback(e);
-    }
-
-    pending = tokens.length;
-
-    var done = function(err) {
-      if (err) {
-        opt.highlight = highlight;
-        return callback(err);
-      }
-
-      var out;
-
-      try {
-        out = Parser.parse(tokens, opt);
-      } catch (e) {
-        err = e;
-      }
-
-      opt.highlight = highlight;
-
-      return err
-        ? callback(err)
-        : callback(null, out);
-    };
-
-    if (!highlight || highlight.length < 3) {
-      return done();
-    }
-
-    delete opt.highlight;
-
-    if (!pending) return done();
-
-    for (; i < tokens.length; i++) {
-      (function(token) {
-        if (token.type !== 'code') {
-          return --pending || done();
-        }
-        return highlight(token.text, token.lang, function(err, code) {
-          if (err) return done(err);
-          if (code == null || code === token.text) {
-            return --pending || done();
-          }
-          token.text = code;
-          token.escaped = true;
-          --pending || done();
-        });
-      })(tokens[i]);
-    }
-
-    return;
-  }
+function marked(src, opt) {
```

Comment:
I suppose it's unlikely we'll unfork this module anytime soon, but it seems like typing the error callback parameter as `any` would be a lot cleaner than forking this further. I'm also kinda skeptical that this patch doesn't break the module? Not all of that code is related to the error callback...

Reply:
I was also skeptical, but we definitely never use the callback so I guess that code was never running? We only used marked in one place so it's hard to imagine what else could be going on. I'm happy to keep the callback if you prefer that though

Reply:
Oh actually -- one reason I removed it was it makes the typing situation weirder for `opt`.

> ### callback
>
> Type: `function`
>
> Function called when the `markdownString` has been fully parsed when using
> async highlighting. If the `options` argument is omitted, this can be used as
> the second argument.

So `opt` would be typed as the the options OR a function/any, which feels undesirable to me as far as typing goes, and we never use it, so I removed the option to have the callback at all.

Would you still prefer me to keep the callback? Happy to add it back if you still want it, after that added context.

Reply:
We don't need to export a completely correct type; we can export a narrower type that pretends their weird callback-as-second-argument feature (which we're not using) doesn't exist.

Reply:
Okay, I'll remove this commit then and update the stub to this:

```js
declare const marked: {
    // Note: We don't use the `callback` option in any of our code.
    (src: string, opt: MarkedOptions, callback?: any): string;
```